spec/filters/geoip_spec.rb

```diff
@@ -221,6 +221,23 @@
         subject
       end
     end
+  end
+
+  describe "returned object identities" do
+    let(:plugin) { LogStash::Filters::GeoIP.new("source" => "message") }
+    let(:geo_data) { plugin.get_geo_data_for_ip("8.8.8.8") }
+
+    before do
+      plugin.register
+    end
+
+    it "should dup the objects" do
+      event = {}
+      alt_event = {}
+      plugin.apply_geodata(geo_data, event)
+      plugin.apply_geodata(geo_data, alt_event)
 
+      event["geoip"].each {|k,v| expect(v).not_to eql(alt_event[k]) }
```

Comment:
But this is a good question I think rspec's eql is checking `object_id` for that specific case.